pkg/apis/kops/v1alpha1/cluster.go

```diff
@@ -251,6 +254,24 @@ type ClusterSpec struct {
 	Assets *Assets `json:"assets,omitempty"`
 }
 
+// FileAssetSpec defines the structure for a file asset
+type FileAssetSpec struct {
+	// Name is a shortened reference to the asset
+	Name string `json:"name,omitempty"`
+	// Path is the location this file should reside
+	Path string `json:"path,omitempty"`
+	// Mode is unix filemode for the file - defaults to 0400
+	Mode string `json:"mode,omitempty"`
```

Comment:
Do you need the mode?  I'm more worried about execute permissions and noexec than anything else here...